.github/workflows/scripts/bump-versions.sh

```diff
@@ -0,0 +1,144 @@
+# Function to display usage
+usage() {
+  echo "Usage: $0 --current_beta <current_beta> [--current_stable <current_stable>] [--next_beta <next_beta>] [--next_stable <next_stable>]"
+  exit 1
+}
+
+# Function to validate semantic version and strip leading 'v'
+validate_and_strip_version() {
+  local var_name=$1
+  local version=${!var_name}
+  # Strip leading 'v' if present
+  version=${version#v}
+  if [[ ! $version =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo "Invalid version: $version. Must be a semantic version (e.g., 1.2.3)."
+    exit 1
+  fi
+  eval "$var_name='$version'"
+}
+
+# Parse named arguments
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --current_beta) current_beta="$2"; shift ;;
+    --current_stable) current_stable="$2"; shift ;;
+    --next_beta) next_beta="$2"; shift ;;
+    --next_stable) next_stable="$2"; shift ;;
+    *) echo "Unknown parameter passed: $1"; usage ;;
+  esac
+  shift
+done
+
+# Check if at least one of the required arguments is provided
+if [ -z "$current_beta" ] && [ -z "$current_stable" ]; then
+  usage
+fi
+
+# Validate and strip versions
+if [ -n "$current_beta" ]; then
+  validate_and_strip_version current_beta
+fi
+if [ -n "$current_stable" ]; then
+  validate_and_strip_version current_stable
+fi
+if [ -n "$next_beta" ]; then
+  validate_and_strip_version next_beta
+fi
+if [ -n "$next_stable" ]; then
+  validate_and_strip_version next_stable
+fi
+
+# Function to bump the minor version and reset patch version to 0
+bump_version() {
+  local version=$1
+  local major
+  major=$(echo "$version" | cut -d. -f1)
+  local minor
+  minor=$(echo "$version" | cut -d. -f2)
+  local new_minor
+  new_minor=$((minor + 1))
+  echo "$major.$new_minor.0"
+}
+
+# Infer the next beta version if not supplied
+if  [ -n "$current_beta" ] && [ -z "$next_beta" ]; then
+  next_beta=$(bump_version "$current_beta")
+fi
+
+# Infer the next stable version if current_stable provided and next version not supplied
+if [ -n "$current_stable" ] && [ -z "$next_stable" ]; then
+  next_stable=$(bump_version "$current_stable")
+fi
+
+
+# List of files to update
+# TODO: Uncomment cmd/builder/builder-config.yaml once PR #671 merged
+files=(
+  "distributions/otelcol-contrib/manifest.yaml"
+  "distributions/otelcol/manifest.yaml"
+  "distributions/otelcol-k8s/manifest.yaml"
+  "distributions/otelcol-otlp/manifest.yaml"
+  "Makefile"
+  # "cmd/builder/builder-config-yaml"
+)
+
+# Update versions in each file
+for file in "${files[@]}"; do
+  if [ -f "$file" ]; then
+    if [ -n "$current_beta" ]; then
+      sed -i.bak "s/$current_beta/$next_beta/g" "$file"
+      echo "Updated $file from $current_beta to $next_beta"
+    fi
+    if [ -n "$current_stable" ]; then
+      sed -i.bak "s/$current_stable/$next_stable/g" "$file"
+      echo "Updated $file from $current_stable to $next_stable"
+    fi
+    rm "${file}.bak"
+  else
+    echo "File $file does not exist"
+  fi
+done
+
+echo "Version update completed."
+
+Commit changes and draft PR
```

Comment:
As mentioned in https://github.com/open-telemetry/opentelemetry-collector-releases/pull/684#pullrequestreview-2345254646:

> Committing changes should be separate from the script, so that anyone can run the script locally and preview the changes before making it a commit.

At the very least, the commit, push, PR creation part should be opt-in via a script option like `--commit` or similar. The default (when the `--commit` option is not passed to the script) should be to not make the commit.

Reply:
added two flags, --commit which commits changes and creates new branch, pushing to remote, and --pull-request which creates draft PR with that new branch. Github workflow passes --commit every time, but allows release manager to check or uncheck --pull-request (checked by default).